pkg/Library/NodeGraph/ConnectionUpdator.js

```diff
@@ -17,15 +17,17 @@ update(inputs, state) {
     const {pipeline, nodeTypes, candidates} = inputs;
     assign(state, {pipeline, candidates});
     if (candidates) {
-      //let changed = false;
```

Comment:
maybe 

let changed = this.removePipeline...
changed ||= this.updatePiepline...

Reply:
`change || = doSomething()` will not call `doSomething`, if `changed` is already `true`.

Reply:
Touché